libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -186,12 +186,18 @@ func (n *bridgeNetwork) addPortMappings(
 			if !ok {
 				return nil, fmt.Errorf("invalid child host IP address %s in %s", b.childHostIP, b)
 			}
-			b.portDriverRemove, err = pdc.AddPort(ctx, b.Proto.String(), hip, chip, int(b.HostPort))
+			bindings[i].portDriverRemove, err = pdc.AddPort(ctx, b.Proto.String(), hip, chip, int(b.HostPort))
```

Comment:
This can still be `b`?

Reply:
Before this change, `b` was a pointer to a binding. Now it's just a shallow copy. Instead of dereferencing `b` (or accessing the slice item directly) when calling network methods below, `b` is used everywhere and `bindings[i]` is used only when overwriting something.

Reply:
Oh yes, thank you - I'd missed that it's not a pointer any more.